plugins/module_utils/linode_event_poller.py

```diff
@@ -130,3 +130,35 @@ def poll_func():
         )
 
         return event
+
+
+def wait_for_resource_free(client: LinodeClient, entity_type: str, entity_id: int, timeout: int):
+    """
+    Waits for all events relevant events to not be scheduled or in-progress.
+    """
+
+    timeout_ctx = TimeoutContext(timeout_seconds=timeout)
+
+    api_filter = {
+        '+order': 'desc',
+        '+order_by': 'created',
+        'entity.id': entity_id,
+        'entity.type': entity_type,
+    }
+
+    def poll_func():
+        events = client.get('/account/events', filters=api_filter)['data']
+        for event in events:
+            if event['status'] in ('scheduled', 'started'):
+                return False
+
+        return True
```

Comment:
Just a fancy way to write it, lol. Feel free to ignore it.

```suggestion
return all(event['status'] not in ('scheduled', 'started') for event in events)
```

Reply:
Nice!